Assets/MixedRealityToolkit.SDK/Experimental/Features/Support/Scripts/BaseServiceManager.cs

```diff
@@ -0,0 +1,395 @@
+
+            registeredServices.Remove(interfaceType);
+            return MixedRealityServiceRegistry.RemoveService<T>(serviceInstance, this);
+        }
+
+        /// <summary>
+        /// Activates an instance of the specified concrete type using the provided argument collection.
+        /// </summary>
+        /// <typeparam name="T">The interface which must be implemented by the concrete type.</typeparam>
+        /// <param name="concreteType">The type of object to be instantiated.</param>
+        /// <param name="supportedPlatforms">The platform(s) on which the concrete type is supported.</param>
+        /// <param name="args">Collection of arguments to provide to the concrete type's constructor.</param>
+        /// <returns>An instance of the concrete type. Returns a default value of T (typically null) in the event of a failure.</returns>
+        private T ActivateInstance<T>(Type concreteType, SupportedPlatforms supportedPlatforms = (SupportedPlatforms)(-1), params object[] args) where T : IMixedRealityService
+        {
+            if (concreteType == null) { return default(T); }
+
+#if UNITY_EDITOR
+            if (!UnityEditor.EditorUserBuildSettings.activeBuildTarget.IsPlatformSupported(supportedPlatforms))
+#else
+            if (!Application.platform.IsPlatformSupported(supportedPlatforms))
+#endif
+            {
+                return default(T);
+            }
+
+            if (!typeof(T).IsAssignableFrom(concreteType))
+            {
+                Debug.LogError($"Error: {concreteType.Name} service must implement {typeof(T)}.");
+                return default(T);
+            }
+
+            try
+            {
+                T serviceInstance = (T)Activator.CreateInstance(concreteType, args);
+                return serviceInstance;
+            }
+            catch (Exception e)
+            {
+                Debug.LogError($"Error: Failed to instantiate {concreteType.Name}: {e.GetType()} - {e.Message}");
+                return default(T);
+            }
+        }
+
+        #endregion IMixedRealityServiceRegistrar implementation
+
+        /// <summary>
+        /// Initialize a service.
+        /// </summary>
+        /// <typeparam name="T">The interface type for the service to be intialized.</typeparam>
+        /// <param name="concreteType">The concrete type of the service to initialize.</param>
+        /// <param name="supportedPlatforms">The platoform(s) on which the service is supported.</param>
+        /// <param name="args">Arguments to provide to the service class constructor.</param>
+        protected virtual void Initialize<T>(Type concreteType, SupportedPlatforms supportedPlatforms = (SupportedPlatforms)(-1), params object[] args) where T : IMixedRealityService
+        {
+            RegisterService<T>(concreteType, supportedPlatforms, args);
+            T serviceInstance = FindService<T>();
+            serviceInstance?.Initialize();
```

Comment:
Should we look at the returned value of RegisterService? No need to FindService if it didn't succeed.

Also, should we log in the case that RegisterService fails?

Reply:
FindService gives us the instance that was created by this overload of RegisterService,

Reply:
fixed